vital/iterator.h

```diff
@@ -160,26 +160,43 @@ class base_iterator
    *
    * \returns If the two iterators are equal to each other.
    */
-  friend bool operator==( base_iterator<T, Tb> const & lhs,
-                          base_iterator<T, Tb> const & rhs )
+  virtual
+  bool equal_to( base_iterator<T, Tb> const & other ) const
```

Comment:
Would it make sense to make this private or protected? I could see it as redundant or confusing if there are multiple avenues exposed for checking equality.

Reply:
I'm fine with making it protected, the derived class needs to assess it to get around another ambigious overload of operator==